File: mcspring-build/mcspring-plugin-manager/src/main/java/in/kyle/mcspring/manager/commands/CommandHeal.java
package in.kyle.mcspring.manager.commands;

import org.bukkit.entity.Player;
import org.springframework.stereotype.Component;

import in.kyle.mcspring.command.Command;
import in.kyle.mcspring.subcommands.PluginCommand;

@Component
class CommandHeal {
    
    @Command(value = "heal",
             description = "Heal yourself or another player",
             usage = "/heal <player>?")
    void heal(PluginCommand command) {
        command.withPlayerSender("Sender must be a player");
        command.withPlayer(s -> String.format("Player %s not found", s));
        command.then(this::executeHeal);
        command.otherwise(this::executeHeal);
    }
    
    String executeHeal(Player target) {
        target.setHealth(target.getMaxHealth());
        return String.format("Healed %s", target.getName());
    }
}

File: mcspring-build/mcspring-plugin-manager/src/main/kotlin/in/kyle/mcspring/manager/commands/CommandHeal.kt
package `in`.kyle.mcspring.manager.commands

import `in`.kyle.mcspring.command.Command
import `in`.kyle.mcspring.commands.dsl.commandExecutor
import `in`.kyle.mcspring.commands.dsl.otherwise
import org.bukkit.entity.Player
import org.springframework.stereotype.Component

@Component
internal class CommandHeal {

    @Command(
            value = "heal",
            description = "Heal yourself or another player",
            usage = "/heal <player>?"
    )
    fun heal() = commandExecutor {
        requirePlayer { message("Sender must be a player") }

        val target by playerArg {
            default { sender as Player }
            invalid { message("Target player $it not found") }
        }

        val health by doubleArg {
            default { 20.0 }
            parser {
                between(0.0, 20.0) otherwise { message("Heal value must be between 0 and 20") }
            }
            invalid { message("Invalid health amount $it") }
        }

        then { message(executeHeal(target, health)) }
    }

    private fun executeHeal(target: Player, health: Double): String {
        @Suppress("DEPRECATION")
        target.health += health
        return "Healed ${target.name} by $health"
    }
}
